File: SUMIT/CPP/ARRAY/TWO_POINTERS/REMOVES DUPLICATES FROM AN ARRAY/Lgieetcode27.cpp
//REMOVE A VALUE WITH ITS DUPLICATES FROM AN ARRAY

#include<bits/stdc++.h>
using namespace std;
// LEETCODES 27
// USING THE CONCEPTS OF TWO POINTERS
// TC ==>0(N)
// SC ==> 0(1)


   int removesDuplicates(vector<int>array, int target) {
    int n = array.size();
    if (n == 0)
      return 0; // EDGE CASE
    int count = 0;
    for (int i = 0; i < n; i++) {
      if (array[i] != target) {
        array[count] = array[i];
        count++;
      }
    }
    return count;
  }

  int main() {
    vector<int>array = { 2, 3, 4, 4, 4, 5, 6, 6, 7, 7, 7, 8, 9, 9, 0, 0 };
    int target = 0;
    int len = removesDuplicates(array,target);

    cout<<"After removing the target value with its duplicates the length of the array is :"<< len;
    for (int i = 0; i < len; i++) {
      cout<< array[i] << " ";
    }
  }

File: SUMIT/JAVA/ARRAY/TWO_POINTERS/REMOVES DUPLICATES FROM AN ARRAY/Lgieetcode27.java

//REMOVE A TARGET VALUES WITH THEIR DUPLICATES FROM AN ARRAY
import java.util.*;
// LEETCODES 27
// USING THE CONCEPTS OF TWO POINTERS
// TC ==>0(N)
// SC ==> 0(1)

public class Lgieetcode27

{
    static int removesDuplicates(int []array , int target){
        int n = array.length;
        if (n == 0) return 0; //EDGE CASE
        int count = 0;
        for (int i = 0; i < n; i++) {
            if (array[i] != target) {
                array[count] = array[i];
                count++;
            }
        }
        return count;
    }
    
    
	public static void main(String[] args) {
	    int[]array = {2,3,4,4,4,5,6,6,7,7,7,8,9,9,0,0};
	    int target = 0;
	    int len = removesDuplicates(array , target); 
	    System.out.println("After removing the duplicates th elength of the array is : "+ len);
	    for(int i = 0 ; i<len ; i++){
		System.out.print( array[i] + " ");
	    }
	}
}